File: IQ/Classes/clsServiceType.vb
Option Explicit On
Option Strict On

Imports dataAccess
Imports System.IO

' Represents an HPE Care Pack service type level

Public Class clsServiceType
    Implements i_Editable

    Public ID As Integer
    Public mfrCode As String
    Public Title As clsTranslation
    Public Description As clsTranslation
    Public ServiceTypeDefault As Boolean

    Private Const TABLE As String = "ServiceType"

    Public Sub New()

    End Sub

    Public Sub New(id As Integer, mfrCode As String, title As clsTranslation, description As clsTranslation, serviceTypeDefault As Boolean)

        Me.ID = id
        Me.mfrCode = mfrCode
        Me.Title = title
        Me.Description = description
        Me.ServiceTypeDefault = serviceTypeDefault

    End Sub

    Public Function Insert(ByRef errormessages As List(Of String)) As Object Implements i_Editable.Insert

        Return New clsServiceType(Me.ID, Me.mfrCode, Me.Title, Me.Description, Me.ServiceTypeDefault)

    End Function

    Public Sub Delete(ByRef errorMessages As List(Of String)) Implements i_Editable.delete

        Dim sql$ = String.Format("delete from {0} where id={1}", TABLE, Me.ID)

        da.DBExecutesql(sql$)

    End Sub

    Public Sub Update(ByRef errorMessages As List(Of String)) Implements i_Editable.update

        Dim sql$ = String.Format("update {0} set mfrCode='{1}', IsDefault={2} where ID={3}", TABLE, Me.mfrCode, Me.ServiceTypeDefault, Me.ID)

        da.DBExecutesql(sql$, False)

    End Sub

    Public Function DisplayName(language As clsLanguage) As String Implements i_Editable.displayName

        Return Me.Title.text(language)

    End Function

End Class

File: IQ/Classes/clsServiceType.cs
using dataAccess;
using System.IO;



// Represents an HPE Care Pack service type level

public class clsServiceType : i_Editable
{

    public int ID;
    public string mfrCode;
    public clsTranslation Title;
    public clsTranslation Description;
    public bool ServiceTypeDefault;

    private const string TABLE = "ServiceType";

    public clsServiceType()
    {

    }

    public clsServiceType(int id, string mfrCode, clsTranslation title, clsTranslation description, bool serviceTypeDefault)
    {

        this.ID = id;
        this.mfrCode = mfrCode;
        this.Title = title;
        this.Description = description;
        this.ServiceTypeDefault = serviceTypeDefault;

    }

    public dynamic Insert(ref List<string> errormessages)
    {

        return new clsServiceType(this.ID, this.mfrCode, this.Title, this.Description, this.ServiceTypeDefault);

    }

    public void delete(ref List<string> errorMessages)
    {

        System.Char sql = string.Format("delete from {0} where id={1}", TABLE, this.ID);

        da.DBExecutesql(sql);

    }

    public void update(ref List<string> errorMessages)
    {

        System.Char sql = string.Format("update {0} set mfrCode=\'{1}\', IsDefault={2} where ID={3}", TABLE, this.mfrCode, this.ServiceTypeDefault, this.ID);

        da.DBExecutesql(sql, false);

    }

    public string displayName(clsLanguage language)
    {

        return this.Title.text(language);

    }

}